Application/Frontend/components/Server/Member_List.tsx

```diff
@@ -51,6 +62,90 @@ export function MemberList({isAdmin}: any) {
       }
   }
 
+  const fetchUserName = async (memberIDs: String[]) => {
+    try {
+      const response = await fetch('/api/ID-to-User-Name', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        //-----------------------------------------------------------------------------------------------------------------------------------------------
+        body: JSON.stringify({ memberIDs: memberIDs}),                              // Change this when we put in the Appshell (It is a String NOT int)
+        //------------------------------------------------------------------------------------------------------------------------------------------------
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        // console.log("My users:" + data.userArray);
+        setMembersList(data.userArray);
+      } else {
+        console.error('Failed to fetch member list');
+      }
+    } catch (error) {
+      console.error('Error fetching member list:', error);
+    }
+  };
+  const addMember = async () => {
+    if (friendUsername.trim()) {
+      try {
+        const response = await fetch('/api/add-member', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({ channelKey: channelKey ,friendUsername: friendUsername }),
+        });
+
+        setSearchResult(response.status);
+
+        if (response.ok) {
+          setFriendUsername('');
+          // Use Mantine's showNotification for success message
+          showNotification({
+            title: 'Success',
+            message: `${friendUsername} was added to the chats!`,
+            color: 'green',
+          });
+          const data = await response.json();
+          setMembersList([...membersList, friendUsername])
+          setMembersIDList([...membersIDList, data.memberID]);
```

Comment:
Good use of React states to add the `friendUsername` and `memberID` to the existing `membersList` / `membersIDList`